### DISCUSSION TITLE
Is there a way to redirect all youtube urls to yewtu.be (https://yewtu.be/) ?

### DISCUSSION THREAD

**Discussion by GasparVardanyan** [qutebrowser/qutebrowser#6836] [Q&A] [answered]
For example to have _https://www.youtube.com/watch?v=Ba9xGhTiy2A_ redirected to _https://yewtu.be/watch?v=Ba9xGhTiy2A_ ?

**Comment by toofar** [comment, marked as answer]
We don't really have official docs around this at the moment but there are a couple of methods the community uses that you may consider, some assembly required:
* via  greasemonkey script: https://gist.github.com/bitraid/d1901de54382532a03b9b22a207f0417
* or via config.py and the interceptor API  https://old.reddit.com/r/qutebrowser/comments/n90y93/a_simple_script_to_convert_reddit_links_to/gxqqyne/

**Reply by GasparVardanyan** [reply]
I've applied this in my config:
```python
import qutebrowser.api.interceptor

def rewrite(request: qutebrowser.api.interceptor.Request):
if request.request_url.host() == 'www.youtube.com':
    request.request_url.setHost('yewtu.be')
    try:
        request.redirect(request.request_url)
    except:
        pass

qutebrowser.api.interceptor.register(rewrite)
```

And qutebrowser crashed. This is the log: http://sprunge.us/eufDQo

**Reply by toofar** [reply]
Sorry, I don't know how to help with that. I have found the redirector thing to be a bit flaky too, often I can fix it by removing the redirection, following the link, then re-adding it. But that isn't part of the API currently. Anyway, it isn't particularly reproducible so no-one has raised an issue upstream with a stack trace from a debug build yet.

Guess it is on to the greasemonkey option!

**Reply by GasparVardanyan** [reply]
Thank you very much! Greasemonkey works well!

**Reply by cortsf** [reply]
Just for the record, you need some indentation there. This works for me:


```
import qutebrowser.api.interceptor

def rewrite(request: qutebrowser.api.interceptor.Request):
  if request.request_url.host() == 'www.youtube.com':
      request.request_url.setHost('localhost')
      request.request_url.setPort(3000)
      request.request_url.setScheme("http")
      try:
          request.redirect(request.request_url)
      except:
          pass
          
qutebrowser.api.interceptor.register(rewrite)
```
